### PR TITLE
chore: upgrade go mod dependencies

### PR DESCRIPTION
**Pull request by jbedard**
This reduces the noise a lot when adding more libraries such as in #1908

**What type of PR is this?**

> Other

**What package or component does this PR mostly affect?**

> all

**What does this PR do? Why is it needed?**

**Which issues(s) does this PR fix?**

Fixes #

**Other notes for review**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,17 +1,17 @@
 module github.com/bazelbuild/bazel-gazelle
 
-go 1.21
+go 1.21.1
 
 require (
-	github.com/bazelbuild/buildtools v0.0.0-20240313121412-66c605173954
+	github.com/bazelbuild/buildtools v0.0.0-20240827154017-dd10159baa91
```
Comment (jbedard) [inline go.mod:6]: This update is what deleted so much in go.sum. The `golang.org/*` had no transitive changes other then what is this go.mod.

File: go.mod [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
 	github.com/bazelbuild/rules_go v0.46.0
 	github.com/bmatcuk/doublestar/v4 v4.6.1
 	github.com/fsnotify/fsnotify v1.7.0
 	github.com/google/go-cmp v0.6.0
 	github.com/pmezard/go-difflib v1.0.0
-	golang.org/x/mod v0.16.0
-	golang.org/x/sync v0.6.0
+	golang.org/x/mod v0.20.0
```
Comment (jbedard) [inline go.mod:12]: Note there is a v0.21 but that requires go 1.22.

File: language/proto/known_imports.go [hunk 3]
```diff
@@ -7,16 +7,16 @@ import "github.com/bazelbuild/bazel-gazelle/label"
...
-	"google/protobuf/descriptor.proto":      label.Label{Repo: "com_google_protobuf", Pkg: "", Name: "descriptor_proto"},
-	"google/protobuf/duration.proto":        label.Label{Repo: "com_google_protobuf", Pkg: "", Name: "duration_proto"},
-	"google/protobuf/empty.proto":           label.Label{Repo: "com_google_protobuf", Pkg: "", Name: "empty_proto"},
-	"google/protobuf/field_mask.proto":      label.Label{Repo: "com_google_protobuf", Pkg: "", Name: "field_mask_proto"},
-	"google/protobuf/source_context.proto":  label.Label{Repo: "com_google_protobuf", Pkg: "", Name: "source_context_proto"},
-	"google/protobuf/struct.proto":          label.Label{Repo: "com_google_protobuf", Pkg: "", Name: "struct_proto"},
-	"google/protobuf/timestamp.proto":       label.Label{Repo: "com_google_protobuf", Pkg: "", Name: "timestamp_proto"},
-	"google/protobuf/type.proto":            label.Label{Repo: "com_google_protobuf", Pkg: "", Name: "type_proto"},
-	"google/protobuf/wrappers.proto":        label.Label{Repo: "com_google_protobuf", Pkg: "", Name: "wrappers_proto"},
+	"google/protobuf/any.proto":             label.Label{Repo: "protobuf", Pkg: "", Name: "any_proto"},
```
Comment (fmeum) [inline language/proto/known_imports.go:10]: Is this change intentional? It looks like it could break the WORKSPACE setup - could you extract it out?

Comment (jbedard) [reply language/proto/known_imports.go]: I reverted this for now and the PR is green still, so I guess this isn't required in this PR (I originally had more changes in this PR and I guess didn't rollback everything such as this file).

This change will be required when running the update with go 1.22 which has a lot more changes then what's in this PR. That can be next...

**Review by fmeum — APPROVED** [review]
